python/tchannel/tornado/hyperbahn.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import json
+import random
+import tornado.gen
+
+from .messages.error import ErrorCode
+from .tornado.tchannel import TChannel
+
+
+DEFAULT_TTL = 60  # seconds
+
+
+class HyperbahnRegistrant(object):
+    """Register a service with a Hyperbahn routing mesh."""
+
+    def __init__(self, hyperbahn_client, service):
+        """
+        :param hyperbahn_client: ``HyperbahnClient`` to make ``register``
+            requests with.
+
+        :param service: Name of service to register with Hyperbahn, eg "maps".
+        """
+        self.hyperbahn_client = hyperbahn_client
+        self.service = service
+
+    @tornado.gen.coroutine
+    def register(self):
+        """Enable register heartbeat, adding service to the Hyperbahn."""
+
+        forward_request = dict(
+            service="hyperbahn",
+            endpoint="ad",  # advertise
+            body=json.dumps({
+                "services": [
+                    {
+                        "serviceName": self.service,
+                        "cost": 0,
+                    },
+                ],
+            }),
+            protocol_headers={
+                "as": "json",
+            },
+        )
```

Comment:
> why not use yield ArgSchemeBroker

Because two imports to do this is very awkward and feels like it's leaking an implementation detail. Something like `hyperbahn_client.request(..., argscheme="json")` would be a lot cleaner.